tests/lib/rules/no-template-curly-in-string.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @fileoverview Tests for rule no-template-curly-in-string
+ * @author Pig Fang <g-plane@hotmail.com>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/no-template-curly-in-string");
+const RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 6 } });
+
+ruleTester.run("no-template-curly-in-string", rule, {
+    valid: [
+        "`${a}`"
+    ],
+    invalid: [
+        {
+            code: "'${a}'",
+            output: "`${a}`",
+            errors: [{ type: "Literal" }]
```

Comment:
这个好像修改了原来的语义，如果a未定义的话，会抛出一个错误。

Reply:
虽然确实会抛出错误，但启用这个规则的开发者应该是为了避免用错引号（暂时不考虑他们 *真的* 想在普通字符串里使用 `${}` 的情况，因为这时候他们一般会临时关闭规则）。也就是说，如果他们本来就打算使用字符串模板，那么抛出错误也是合理的。如果在同时开启 `no-undef` 的情况下，未定义还可以被 ESLint 捕捉。

Reply:
> Best practices for fixes:
> 
> Avoid any fixes that could change the runtime behavior of code and cause it to stop working.
> Make fixes as small as possible. Fixes that are unnecessarily large could conflict with other fixes, and prevent them from being applied.
> Only make one fix per message. This is enforced because you must return the result of the fixer operation from fix().
> Since all rules are run again after the initial round of fixes is applied, it’s not necessary for a rule to check whether the code style of a fix will cause errors to be reported by another rule.

Reply:
这个规则有点类似 valid-typeof（fix 后都会影响执行结果）, 但是valid-typeof 绝大数情况下fix后的结果是预期（极少有例外）。

Reply:
我们应该定一条红线：什么情况下不能autofix。

我的建议是：凡是有可能影响执行结果的**原则上**都不能fix，但是可以有例外（case by case)。

Reply:
这类放到 unsafe 里吗？

Reply:
是的。

Reply:
那这个 PR 应该如何处理？

Reply:
关掉? 或者归类到unsafe。。有点纠结

让你来决定吧! 😄